lib/dataloader/kv.ex

```diff
@@ -69,14 +69,25 @@ defmodule Dataloader.KV do
     end
 
     def load(source, batch_key, id) do
-      case fetch(source, batch_key, id) do
-        {:error, _message} ->
+      unless fetched?(source.results, batch_key, id) do
```

Comment:
If you're going to have an `else` I vastly prefer just inverting the flow:

```
if fetched?() do
  source
else
  update_in()
end
```
```

Reply:
makes sense